dev/udf_proxy.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <sqlite3.h>
+#include <new>  //  std::launder
+#include <string>  //  std::string
+#include <functional>  //  std::function
+#include <utility>  //  std::move
+
+#include "error_code.h"
+
+namespace sqlite_orm {
+    namespace internal {
+        /*
+         *  Stores type-erased information in relation to a user-defined scalar or aggregate function object:
+         *  - name and argument count
+         *  - function pointers for construction/destruction
+         *  - function dispatch
+         *  - allocated memory and location
+         *  
+         *  As such, it also serves as a context for aggregation operations instead of using `sqlite3_aggregate_context()`.
+         */
+        struct udf_proxy {
+            using func_call_fn_t = void (*)(void* udfHandle,
+                                            sqlite3_context* context,
+                                            int argsCount,
+                                            sqlite3_value** values);
+            using final_call_fn_t = void (*)(void* udfHandle, sqlite3_context* context);
+
+            struct destruct_only_deleter {
+                template<class F>
+                void operator()(F* f) const noexcept {
+                    f->~F();
+                }
+            };
+
+            std::string name;
+            int argumentsCount;
+            std::function<void*(void* location)> constructAt;
+            xdestroy_fn_t destroy;
+            func_call_fn_t func;
+            final_call_fn_t finalAggregateCall;
+            // flag whether the UDF has been constructed at `udfHandle`;
+            // necessary for aggregation operations
+            bool constructed;
+            // pointer to memory for UDF in derived proxy veneer
+            void* const udfHandle;
+        };
+
+        /*
+         *  A veneer to `udf_proxy` that provides memory space for a user-defined function object.
+         *  
+         *  Note: it must be a veneer, i.e. w/o any non-trivially destructible member variables.
+         */
+        template<class UDF>
+        struct udf_proxy_veneer : udf_proxy {
+            // allocated memory for user-defined function
+            alignas(UDF) char storage[sizeof(UDF)];

```

Comment:
from the first sight `storage` reminds of `storage_t` but it is a whole different entity type. Probably we better rename it to something that has no existing associations in the lib like `bytes`

Reply:
I have changed the way to allocate memory anyway, with commit [2c23449](https://github.com/fnc12/sqlite_orm/pull/1248/commits/2c234491ccfe05111656110bb06c1e0306a438f9).